src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -187,10 +187,9 @@ def login_with_service_principal(self, client_id, credential, scopes):
         `credential` is a dict returned by ServicePrincipalAuth.build_credential
         """
         sp_auth = ServicePrincipalAuth.build_from_credential(self.tenant_id, client_id, credential)
-
-        # This cred means SDK credential object
-        cred = ServicePrincipalCredential(sp_auth, **self._msal_app_kwargs)
-        result = cred.acquire_token_for_client(scopes)
+        client_credential = _build_msal_client_credential(sp_auth)
+        cca = ConfidentialClientApplication(client_id, client_credential=client_credential, **self._msal_app_kwargs)
+        result = cca.acquire_token_for_client(scopes)
```

Comment:
Because `ServicePrincipalCredential` stops inheriting from `msal.ConfidentialClientApplication`, it is no longer possible to call `acquire_token_for_client` on the `cred`. We prepare `client_credential` and directly create a `ConfidentialClientApplication` instance.